src/map/lua/lua_baseentity.cpp

```diff
@@ -366,6 +366,106 @@ inline int32 CLuaBaseEntity::isJugPet(lua_State* L)
 
 //======================================================//
 
+inline int32 CLuaBaseEntity::getCharmChance(lua_State* L)
+{
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isuserdata(L, 1));
+
+    CLuaBaseEntity* PLuaBaseEntity = Lunar<CLuaBaseEntity>::check(L, 1);
+    
+    auto PCharmer = dynamic_cast<CBattleEntity*>(m_PBaseEntity);
```

Comment:
Make a function in battleutils that gets the charm chance and have both
call that